automatic/nuget.commandline/update.ps1

```diff
@@ -19,21 +19,30 @@ function global:au_BeforeUpdate {
 function global:au_GetLatest {
   $json = Invoke-WebRequest -UseBasicParsing -Uri $releases | ConvertFrom-Json
 
-  $versions = $json."nuget.exe" | Sort-Object uploaded -Descending
-  $preRelease = $versions | ? stage -EQ 'EarlyAccessPreview' | select -First 1
-  $stableRelease = $versions | ? stage -EQ 'ReleasedAndBlessed' | select -First 1
-
-  $streams = @{
-    'pre'    = @{
-      Version = $preRelease.version
-      URL32   = $preRelease.url
-    }
-    'stable' = @{
-      Version = $stableRelease.version
-      URL32   = $stableRelease.url
+  $versions = $json."nuget.exe"
+
+  $streams = @{}
+
+  $versions | Sort-Object uploaded -Descending | % {
+    $versionTwoPart = $_.version -replace '^(\d+\.\d+).*$','$1'
+
+    if (!$streams.ContainsKey("$versionTwoPart")) {
+      $streams.Add($versionTwoPart, @{
+        Version = $_.Version
+        URL32   = $_.url
+      })
     }
   }
 
+  $preKey = $streams.Keys | ? { $_ -match '-' } | sort -Descending | select -First 1
+  $stableKey = $streams.Keys | ? { $_ -notmatch '-' } | sort -Descending | select -First 1
+  if ($preKey) {
+    $streams.Add('pre', $streams[$preKey])
+    $streams.Remove($preKey)
+  }
+  $streams.Add('stable', $streams[$stableKey])
+  $streams.Remove($stableKey)
```

Comment:
The idea is to have a commonly named stream if we want to push a fix to only the latest stable release (the pre stream will only work if there isn't a later stable version though).

There isn't really a need to check for the state, as a check a few lines up checks if the version contains a `-`, if so it is a pre stream, otherwise a stable stream.

Reply:
Of course, I could change the script up to also check for the `state` in the json file, but from looking at it, it seemed unnecessary.

Reply:
The question is what is the latest stable release. From a NuGet persepective it is the release with the highest version number which has a state of ReleasedAndBlessed. But if we ignore the state and just publish anything, then this makes sense from a Chocolatey Package distribution point of view.

Reply:
From a package perspective, the latest release would be any release not using a pre-release tag. As of right now, that would be `5.0.0`.

Let me think about this a little, perhaps you are onto something here. I could perhaps make some changes that would use the following streams as well `Released`, `Early` and `Blessed`, which would then be set to the latest version in that stage (Considering there is no blessed version higher than release version, and no blessed or release version higher than Early version)

Reply:
@AdmiringWorm just wanted to touch base on this, as an issue was just raised on the choco repository about a missing package version in the 4.9.x stream.  What are your latest thoughts?

Reply:
I know why the latest 4.9.x version wasn't picked up (or a possible reason).

There is a bug in the existing logic of the updater (it is the same logic in this PR, but that don't matter with this, and isn't a bug any longer).

Basically, there was 3 or more version releases on the same day, this meant that only 1 of those releases would be picked up, and unfortunately the latest 4.9.x version was not that version.

When it comes to this PR itself, I do believe I will be updating it with a blessed stream as well.
But that can come later, if there is a hurry to get out the 4.9.4, or even 5.0.0 release.

Reply:
Yes, that was the case. Therefore my original idea to have a stream per version which should have picked up all 3 versions in that case.

Reply:
@pascalberger however, that is no longer needed with the changes that I made (splitting every stream with the Major+Minor version, as well as providing the latest release as a stable/pre stream).